### ISSUE TITLE
Discord-Github Workflow

### ISSUE THREAD

**Issue by mindovermiles262** [TheOdinProject/discussions#11]
~~Placeholder for the Discord-Github Workflow Project on TOP/Projects Kanban.~~
~~Discussion on Forum~~

We get a lot of ideas that come in from users on Discord. There should be a way to integrate these ideas that come up on Discord to our Github where we can create issues, track and manage them.

I propose a workflow as follows:
* User submits great idea in Discord `#contribution-suggestions-bugs-discussions` channel
* Mod approves great idea => Reacts with specific emoji
* Discord bot is listening to channel for specific emoji:
  * If emoji posted by a mod => Discord bot makes API request to Github. Issue is created
  * If emoji not posted by mod => Discord bot ignores.

On the opposite side, we should also have a way of pushing updates in github back to the discord channel. This will keep us interested and informed of the progress. Suggested workflow as follows:

* Issue is updated
* Github actions or webhooks are used to feed update back into `#contribution-suggestions-bugs-discussions` channel

**Comment by ZachBaird** [comment]
I think this would be great to start thinking about pushing soon. We have the Github notifications in hooks at the moment, I think it'd be very useful to start working on the Discord -> Github issue workflow.

**Comment by I3uckwheat** [comment]
@ZachBaird There's a lot that needs to be done with the bot right now. ~~But it for sure will be added soon.~~

Once we get the tests done as well as a refactor.

**Comment by codyloyd** [comment]
I gotta say....... i'm not going to shoot you all down if you wanna try to make it work.. but I do not really like this idea.  IMO it's a pretty slim chance that turning a discord comment into a github issue is going to be actually useful.... like you're gonna have to come in and clean it up anyway, format it correctly, add a title etc.   

At which point you probably should have just started with the issue in GH.

BUT => it's not like it'll hurt anything either, so again, if you guys want to give it a go, have fun.

**Comment by I3uckwheat** [comment]
> I gotta say....... i'm not going to shoot you all down if you wanna try to make it work.. but I do not really like this idea. IMO it's a pretty slim chance that turning a discord comment into a github issue is going to be actually useful.... like you're gonna have to come in and clean it up anyway, format it correctly, add a title etc.
> 
> At which point you probably should have just started with the issue in GH.
> 
> BUT => it's not like it'll hurt anything either, so again, if you guys want to give it a go, have fun.

I get your point, but we can just type up the MD in the discord. I just think it'd be a quicker way to get issues into GH. However, you are right when it comes to tagging and such. That could be a major problem

**Comment by codyloyd** [comment]
> but we can just type up the MD in the discord

lol ok... that doesn't seem useful.. but 🤷

**Comment by I3uckwheat** [comment]
@codyloyd The hope was that we could get issues into GH right away so it's not forgotten about. A copy-paste of a student issue is easier than going to GH and making one. 

That's the intent anyway.

**Comment by CouchofTomato** [comment]
I thought we wanted to keep issues as clear as possible from thrown around ideas. I wouldn’t mind it for bug fixes but I’m not sure it would work well. This would be abused with students throwing ideas around all over the place. This could be a nightmare to manage all the issues created.

**Comment by I3uckwheat** [comment]
@CouchofTomato The idea would be to only let maintainers+ add issues.

**Comment by codyloyd** [comment]
I don't think it would be _that_ bad... so my perspective is: if y'all wanna try it, have fun.

I just can't see it being useful (to me, personally, considering the types of workflows i generally enjoy etc. etc.)

**Comment by codyloyd** [comment]
maybe just dropping this link in the chat would be useful 🤷 https://github.com/TheOdinProject/discussions/issues/new

edit: not the discussions one.. you know what i mean

**Comment by CouchofTomato** [comment]
I don’t personally have an issue opening an issue on GitHub but as long as it’s not open to abuse I don’t care either way. If it can be pulled off it’s a good project for someone.

**Comment by I3uckwheat** [comment]
I'm convinced to leave it as-is at this point. But i'm still cool with someone doing this if they want. *I'd* use it :D

**Comment by codyloyd** [comment]
oh hey


you can create and link to a new issue with querystrings.... _this_ could actually be useful.  Make the bot spit this out with the right info... again with the correct repo:

https://github.com/TheOdinProject/discussions/issues/new?title=New+bug+report&body=issue+body

deets: https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters#supported-query-parameters

**Comment by KevinMulhern** [comment]
A bot command for providing a link to the new issue page would be pretty neat, I can see that being useful. 

And it satisfies the essence of this proposal imo.

**Comment by I3uckwheat** [comment]
Agreed, this made it a real good feature now instead of just a handy one.

**Comment by ZachBaird** [comment]
I agree with all the preceding thoughts, and I'm glad some discussion got under way and we got some use here! 😃
